src/kubernetes.cc

```diff
@@ -1034,8 +1046,11 @@ void KubernetesReader::WatchNode(MetadataUpdater::UpdateCallback callback)
   LOG(INFO) << "Watch thread (node) exiting";
 }
 
-void KubernetesUpdater::start() {
-  PollingMetadataUpdater::start();
+bool KubernetesUpdater::start() {
+  if(!PollingMetadataUpdater::start()) {
+    return false;
+  }
+
   if (config().KubernetesUseWatch()) {
```

Comment:
Does it make sense to use both watch and polling? If so, should we check here whether both are set?

Reply:
I think it makes sense to choose one or the other, but then I'm not sure if the contract of the KubernetesUpdater gets a little blurred, as it extends a PolliingMetadataUpdater, but only acts on that functionality conditionally. It's almost as though there should be a KubernetesPollingUpdater, and a KubernetesWatchUpdater which is controlled by configuration when the updaters are all started. WDYT?

Reply:
I've considered it, but there's an advantage of sharing a reader.
Perhaps the right approach is to factor out a poller that takes a callback, rather than embedding the contract into the hierarchy (i.e., composition over inheritance).

Reply:
Can we run with it as-is for now and refactor in the future?